Update local dev setup and fix iD for OAuth2

geohacker opened the pull request:
This PR:
* updates the local setup — simplifies the docker compose so it's easy to stand up a local env quickly
* fix iD assets to work with oauth2 changes upstream

cc @danrademacher @batpad @Rub21

geohacker commented on README.md:
```diff
@@ -49,23 +49,38 @@ More details on contributing to the code are in the [CONTRIBUTING.md](CONTRIBUTI
+
+1. Check out this repository to your local machine. Make sure you also have Docker installed. You can get Docker at https://docs.docker.com/get-docker/
+
+2. Make a new file at `config/storage.yml` with the following contents:
```
@danrademacher this should just be creating a copy of `example.storage.yml` https://github.com/OpenHistoricalMap/ohm-website/blob/staging/config/example.storage.yml — if it helps we can create these local files as part of the docker build.

danrademacher replied:
I updated the README to suggest a `cp` of that file into the new file, similar to the other steps.

gregallensworth commented on app/assets/javascripts/index.js:
```diff
@@ -403,21 +354,6 @@ $(document).ready(function () {
   };
 
-  // add the enhanced inspector
-  function addOpenHistoricalMapInspector () {
```
Done. Re-fetch the `fix-id-and-local-dev` branch, where I have re-added the TimeSlider and Inspector components. It seems to be working, though I would like to test Inspector more thoroughly if I can find a record with lots of content. (images, license, wikipedia links, ...)

gregallensworth replied:
A good test of Inspector is the Occidental Hotel   http://localhost:3000/way/198099635

danrademacher commented:
OK, the inspector and timeslider are working well!

Now, darn it, we have previous unknown problems to try to solve:

## User menu in upper right not functioning properly
Clicking in the user menu in upper right doesn't open the menu that it should, also no error in the console:
https://recordit.co/nWE22QHcYA

Going to `/logout` or `/user/displayname` does work, to either logout or edit profile, so this seems to be an error with the menu itself, maybe some code that's swallowing the clicks? Rather than a deeper problem. Unrelated, I have zero clue how my local account (which I made with a username like `dan+localohm`) became `jeffmeyer`, but I'm hoping that's some fluke of importing local data or something... 

## CSS collision in embedded iD  
At a link like http://localhost:3000/edit#background=Bing&map=20.00/47.60192/-122.33331 you can see that for some reason the buttons in iD are picking up CSS from the parent site that is sticking a magnifying glass on each button: 
<img width="85" alt="image" src="https://user-images.githubusercontent.com/1423200/207507886-0e966829-1bec-4f11-b6ee-e642e91c0ca9.png">
<img width="81" alt="image" src="https://user-images.githubusercontent.com/1423200/207508032-206672ed-462f-49ea-bb98-3c0ec70b0e0d.png">

The offending SCSS is here https://github.com/OpenHistoricalMap/ohm-website/blob/fix-id-and-local-dev/app/assets/stylesheets/common.scss#L35-L44
```scss
.icon {
  display: inline-block;
  vertical-align: top;
  width: 20px;
  height: 20px;
  background: transparent image-url("sprite.png") no-repeat 0 0;
  background-image: image-url("sprite.svg");
  text-indent: -9999px;
  overflow: hidden;
}
```
But if we just disable that, then all icons throughout the site get clobbered, like the controls on the main map. 

I am surprised that the iD icon styles are not all name-spaced to avoid collision. Looking at iD CSS, we have the same class name here:
https://github.com/OpenHistoricalMap/iD/blob/staging/css/80_app.css#L439-L443
```css
.icon {
    vertical-align: middle;
    width: 20px;
    height: 20px;
}
```

But I just checked and this does not happen upstream in OSM, where iD has only its own styles and not the parent page styles, and I just checked both upstreams and these CSS styles look the same there, so I am not sure what is going on that is making parent site styles apply in one place (OHM) but not the other place (OSM), even those in theory the relevant bits of code are the same. Must be some other relevant bits that I'm not thinking of that are causing this!

gregallensworth commented:
> User menu in upper right not functioning properly

@geohacker I think that an upstream file was not brought in, but should be: **app/views/layouts/_header.html.erb**

The one here at ohm-website is using HTML for Bootstrap 4 instead of Bootstrap 5. Note the `pl` instead of `pe` and `-right` instead of `-end`

https://github.com/OpenHistoricalMap/ohm-website/blob/fix-id-and-local-dev/app/views/layouts/_header.html.erb#L81-L106

https://github.com/openstreetmap/openstreetmap-website/blob/master/app/views/layouts/_header.html.erb#L85-L109

Looking at https://github.com/OpenHistoricalMap/ohm-website/pull/141/files I do not see layouts/_header as a modified file. If I paste in the BS5 paragraph from OSM into _header.html.erb as the new 85-109 content, the user menu then works. So I am confident that bringing in _header.html.erb will fix the user menu button.

gregallensworth commented:
> CSS collision in embedded iD

This is because the iD editor is being included into the page's HTML, as opposed to loading within a iframe. If iD were running in a iframe then this `.icon` class would not be applied. But since it's included into this page, this site's styles are applied.

@geohacker Any ideas why iD would be loading inside the OHM website's document, rather than in a iframe?


### Screenshot: OHM, `#id-container` is here in this same page

![image](https://user-images.githubusercontent.com/3117633/207705993-c83e1535-6b91-4e2e-a938-2ca3422ed350.png)

### Screenshot: OSM, `#id-container` is in a iframe

![image](https://user-images.githubusercontent.com/3117633/207706325-4b01ff1b-15f2-4a79-95e4-a6580cdfa09d.png)


### Screenshot: OHM Staging, `#id-container` is in a iframe

![image](https://user-images.githubusercontent.com/3117633/207706842-f7be316e-c4e9-41a4-b5bc-50b03e630fa1.png)

danrademacher commented on app/views/site/_id.html.erb:
```diff
@@ -1,10 +1,20 @@
-     data[:url] = id_url(:locale => params[:locale]) %>
-  <%= tag.iframe "", :frameBorder => 0, :id => "id-embed", :class => "id-embed", :allowfullscreen => "", :data => data %>
-</div>
+<!DOCTYPE html>
```
@geohacker Can you explain more about this change? It seems to take iD out of an iframe and instead embed it directly in the page. This causes CSS conflicts in the icons. But before we try to either undo this commit or fix the CSS conflicts, it would be helpful to know why you made this change, in case it was to address some other problem.

geohacker commented:
Ah really nice catch @gregallensworth @danrademacher, thank you! You're right — that was an error I introduced while merging in _id.html.erb, I didn't want to remove the HTML tags. I just fixed it and pushed. Was able to test locally that the CSS issues are gone.

danrademacher commented:
@geohacker thanks for the Id iframe fix! Did you also see this from @gregallensworth?
> Looking at https://github.com/OpenHistoricalMap/ohm-website/pull/141/files I do not see layouts/_header as a modified file. If I paste in the BS5 paragraph from OSM into _header.html.erb as the new 85-109 content, the user menu then works. So I am confident that bringing in _header.html.erb will fix the user menu button.

Sounds like we need to pull in that upstream file, right?

That might break our timeslider and inspector again, since we call code in the header, but those errors are easy to fix and more known to us. So I think better to pull in the upstream and then we can clean up, so we can be sure we have the right Bootstrap version and anything else that might have changed there

geohacker commented:
@danrademacher yes, I think we should bring in that file. I did bring that initially but then left it as I ran into a bunch of other component errors without bringing our changes for the timeslider.
